angstudy/src/app/components/star-rating/star-rating.component.html

```diff
@@ -1,6 +1,13 @@
 <div class="star-rating-container star-rating-animation" fxLayout="row" fxLayoutAlign="start center" >
-    <div *ngFor="let star of stars" [ngClass]="[star.class]" (click)="selectStar(star.id)">
-        <mat-icon>{{star.icon}}</mat-icon>       
-    </div> 
-    {{totalFeedBack}} отзывов.
+    <mat-icon
+        *ngFor="let star of stars index as i"
```

Comment:
```suggestion
        *ngFor="let star of stars; index as i"
```
Лучше отделять основную логику директивы и получения ее данных или прокидывания доп данных